src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -143,6 +142,20 @@ private void VerifyExpression(BoundExpression expression, bool overrideSkippedEx
                 return null;
             }
 
+            public override BoundNode? VisitAssignmentOperator(BoundAssignmentOperator node)
+            {
+                // We're not correctly visiting the right side of object creation initializers when
+                // the symbol is null (such as for dynamic)
+                // https://github.com/dotnet/roslyn/issues/45088
+                if (node.Left is BoundObjectInitializerMember { MemberSymbol: null })
```

Comment:
>if (node.Left is BoundObjectInitializerMember { MemberSymbol: null }) [](start = 16, length = 69)

Why put a workaround here instead of overriding visit method for ```BoundObjectInitializerMember``` node and do what is necessary there? It feels like, when the issue is addressed, that method is going to be adjusted.

Reply:
If the left is a `BoundObjectInitializerMember` with a null symbol, we need to skip visiting the entire right side of the assignment. We can't do the adjustment in the nested visit call.

---
In reply to: [441173553](https://github.com/dotnet/roslyn/pull/42403#discussion_r441173553) [](ancestors = 441173553)